src/engine/map.cxx

```diff
@@ -361,10 +372,93 @@ void Map::highlightNode(const Point &isoCoordinates)
   {
     int index = isoCoordinates.x * _columns + isoCoordinates.y;
 
-    if (index >= 0 && index < _mapNodes.size())
+    if (index >= 0 && index < mapNodes.size())
     {
-      highlitNode = _mapNodes.at(index);
+      highlitNode = mapNodes[index].get();
       highlitNode->getSprite()->highlight(true);
     }
   }
 }
+
+void Map::saveMapToFile(const std::string &fileName)
+{
+  json j = json{{"columns", this->_columns}, {"rows", this->_rows}, {"mapNode", mapNodes}};
+
+  std::ofstream file(fileName);
+  file << j;
+}
```

Comment:
Add `file.close();`.

Reply:
The C++ documentation for fstream says, that it's automatically close in it's destructor when it goes out of scope:
http://www.cplusplus.com/reference/fstream/fstream/close/

> Note that any open file is automatically closed when the fstream object is destroyed.

But i'll add it anyway... better be safe than sorry :)